### PR TITLE
Proposal of a fix for bug #1950

### LINKED ISSUE #1950: Disabled "line too long"(C0301) not working on comment lines
### Steps to reproduce
1. Disable "line too long" warning in a python script (# pylint: disable=c0301)
2. Place a overlong comment line.

### Current behavior
Disabled warning works for normal code lines, however on comment lines still warns.

### Expected behavior
Warning should not be shown.

### Environment
(K)ubuntu 16.04.4
Atom 1.25.0

#### pylint --version output
$ pylint --version
No config file found, using default configuration
pylint 1.5.2, 
astroid 1.4.4
Python 2.7.12 (default, Dec  4 2017, 14:50:18) 
[GCC 5.4.0 20160609]

$ pylint3 --version
No config file found, using default configuration
pylint3 1.8.3, 
astroid 1.6.2
Python 3.5.2 (default, Nov 23 2017, 16:37:01) 
[GCC 5.4.0 20160609]

### PR DESCRIPTION
**Pull request by hippo91**
The bug arises when a module ends with a long commented line.
This is because the line doesn't appear in the AST and thus is absent from Lint.file_state._module_msgs_state. 
The strategy adopted is to check if the line number is above the maximum line number present in Lint.file_state._module_msgs_state and if it is the case the message emission is deactivated if the msgid C0301 is present in Lint.file_state._raw_module_msgs_state.
I added a unit test that should not emit any messages.

Close #1950

### CODE REVIEW

**Comment by hippo91** [comment]
@PCManticore, the Travis CI is failing due to pylint's warning even on file i didn't modify. Should i correct them anyway?

File: pylint/utils.py [hunk 1]
```diff
@@ -394,6 +394,13 @@ def is_message_enabled(self, msg_descr, line=None, confidence=None):
         try:
             return self.file_state._module_msgs_state[msgid][line]
         except KeyError:
+            # Check if the message's line is after the maximum line existing in ast tree.
+            # This line won't appear in the ast tree and won't be referred in
+            # self.file_state._module_msgs_state
+            # This happens for example with a commented line at the end of a module.
+            if (self.file_state.get_effective_max_line_number() and
+                    line > self.file_state.get_effective_max_line_number()):
+                return not msgid in self.file_state._raw_module_msgs_state
```
Comment (PCManticore) [inline pylint/utils.py]: `return msgid not in ...`

File: pylint/utils.py [hunk 2]
```diff
@@ -684,6 +691,12 @@ def iter_spurious_suppression_messages(self, msgs_store):
                 yield 'suppressed-message', line, \
                     (msgs_store.get_msg_display_string(warning), from_)
 
+    def get_effective_max_line_number(self):
+        """Return the maximum line number present in _module_msgs_state."""
+        if self._module_msgs_state:
+            return max([max(d.keys()) for d in self._module_msgs_state.values()])
```
Comment (PCManticore) [inline pylint/utils.py]: Maybe could be refactored to use generator expressions:
`max(max(d) for d in self._module_msgs_state.values))`

File: pylint/utils.py [hunk 3]
```diff
@@ -394,6 +394,13 @@ def is_message_enabled(self, msg_descr, line=None, confidence=None):
         try:
             return self.file_state._module_msgs_state[msgid][line]
         except KeyError:
+            # Check if the message's line is after the maximum line existing in ast tree.
+            # This line won't appear in the ast tree and won't be referred in
+            # self.file_state._module_msgs_state
+            # This happens for example with a commented line at the end of a module.
+            if (self.file_state.get_effective_max_line_number() and
```
Comment (PCManticore) [inline pylint/utils.py]: Maybe save the value so we don't call the function two times?

**Comment by PCManticore** [comment]
@hippo91 Regarding the lint failures, we can do them in a separate PR. Weird that we have these right now though.

**Comment by hippo91** [comment]
@PCManticore thanks for the review!

**Comment by hippo91** [comment]
@PCManticore should i backport this PR to 1.9 or 1.8?

**Pull request merged**